Compiler/BackEnd/Tearing.mo

```diff
@@ -4028,5 +3992,16 @@ algorithm
 
 end recursiveTearingReplace;
 
+protected function getUnassigned
+  input array<Integer> ass;
+  output list<Integer> unassigned={};
+algorithm
+  for i in 1:arrayLength(ass) loop
+    if Dangerous.arrayGetNoBoundsChecking(ass, i)==-1 then
+      unassigned := i::unassigned;
+    end if;
+  end for;
+end getUnassigned;
+
```

Comment:
I used something like that before, and it either created a temporary or I was just confused because libgc returned bad numbers. You can also do `unassigned := listReverse(i for i guard Dangerous.arrayGetNoBoundsChecking(ass, i)==-1 in 1:arrayLength(ass)) `. But basically there should not be any difference in performance between the two.